### DISCUSSION TITLE
关于路由component和redirect有子routes将会坏死

### DISCUSSION THREAD

**Discussion by ByYogi** [umijs/umi#12196] [Q&A] [answered]
component和redirect有子routes， 则routes将失效
![image](https://github.com/umijs/umi/assets/111735598/17095aa4-43dc-4859-a464-2c0e34f806b3)

只有这里是好的
![image](https://github.com/umijs/umi/assets/111735598/72773c89-be51-4774-a061-4a49f1ff3cbc)

**Comment by ByYogi** [comment]
@fz6m 请求帮助

**Comment by fz6m** [comment, marked as answer]
如果有 `redirect` ，umi 就自动把这个路由的组件给变成用于重定向的组件了，所以没法有 `routes` 子路由。

如果有 `component` ，那这个 `component` 就展示在这个层级了，如果还想同时有 `routes` （子路由），就需要在 `component` 组件里用 `<Outlet />` 去展示子路由（来自于 umi 4 使用的 [react router v6](https://reactrouter.com/en/main) 的设定），如果不需要子路由就没必要配置 `routes` ，如果需要嵌套子路由，就自行添加 `<Outlet />` 到父路由，或者你不在父路由添加 `component` 这个字段也可以。
